ir/memory.cpp

```diff
@@ -623,8 +675,8 @@ bool Memory::mayalias(bool local, unsigned bid0, const expr &offset0,
                       unsigned bytes, unsigned align, bool write) const {
   if (local && bid0 >= next_local_bid)
     return false;
-  if (!local && (bid0 >= (write ? num_nonlocals_src : numNonlocals()) ||
-                 bid0 < has_null_block + write * num_consts_src))
+  if (!local && (nonlocal_always_noread(bid0) ||
```

Comment:
this change is not NFC
the function only does `bid < has_null_block`

Reply:
Oops, I fixed it